Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/fragments/PermissionsFragment.kt

```diff
@@ -31,34 +35,65 @@ private val PERMISSIONS_REQUIRED = arrayOf(
         Manifest.permission.CAMERA,
         Manifest.permission.RECORD_AUDIO)
 
+// TODO: Change fragment name in refactor commit. Waiting for merge of CameraActivity and AnalyserActivity.
 /**
  * This [Fragment] requests permissions and, once granted, it will navigate to the next fragment
+ * with the lowest camera settings available.
  */
 class PermissionsFragment : Fragment() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
+        
         if (hasPermissions(requireContext())) {
             // If permissions have already been granted, proceed
-            Navigation.findNavController(requireActivity(), R.id.fragment_container).navigate(
-                    PermissionsFragmentDirections.actionPermissionsToSelector())
+            startCamera()
         } else {
             // Request camera-related permissions
             requestPermissions(PERMISSIONS_REQUIRED, PERMISSIONS_REQUEST_CODE)
         }
     }
 
+    private fun startCamera(){
+        val cameraManager =
+                requireContext().getSystemService(Context.CAMERA_SERVICE) as CameraManager
+
+        val cameraList = enumerateHighSpeedCameras(cameraManager)
+
+        val lowestSetting = findLowestCameraSetting(cameraList)
+
+        Navigation.findNavController(requireActivity(), R.id.fragment_container).navigate(
+                PermissionsFragmentDirections.actionPermissionsFragmentToCameraFragment(
+                        lowestSetting.cameraId, lowestSetting.size.width, lowestSetting.size.height, lowestSetting.fps))
+    }
+    /** Selects the lowest camera fps and the lowest image capture resolution available among them.*/
+    private fun findLowestCameraSetting(cameraList: List<CameraInfo>): CameraInfo {
+        var bestLowestSettingSeen = cameraList.get(0)
```

Comment:
optional: Wonder if you can do this more simply with a Stream:
https://developer.android.com/reference/java/util/stream/Stream
and/or a comparator. Something like:
Comparator<CameraInfo> comparator = Comparator.comparing(s -> s.fps);
comparator = comparator.thenComparing(Comparator.comparing(s -> s.size.width));
comparator = comparator.thenComparing(Comparator.comparing(s -> s.size.height));
return cameraList.stream().sorted(comparator).findFirst().get(); // Alternative is to copy the cameraList (to avoid modifying the original collection) and use .sort() directly on the List.